dvc/repo/experiments/queue/local.py

```diff
@@ -124,22 +125,31 @@ def worker(self) -> "TemporaryWorker":
         )
 
     def spawn_worker(self):
-        from shortuuid import uuid
-
         from dvc_task.proc.process import ManagedProcess
 
         logger.debug("Spawning exp queue worker")
         wdir_hash = hashlib.sha256(self.wdir.encode("utf-8")).hexdigest()[:6]
-        node_name = f"dvc-exp-{wdir_hash}-1@localhost"
+        number = 1
+        node_name = f"dvc-exp-{wdir_hash}-{number}@localhost"
+        worker_status = self.active_worker()
+        while node_name in worker_status:
+            number += 1
+            node_name = f"dvc-exp-{wdir_hash}-{number}@localhost"
+
         cmd = ["exp", "queue-worker", node_name]
-        name = "dvc-exp-worker"
-        if logger.getEffectiveLevel() < logging.INFO:
-            name = name + str(uuid())
+        name = f"dvc-exp-worker-{number}"
+        logger.debug(f"start worker: {name}, node: {node_name}")
         if os.name == "nt":
             daemonize(cmd)
         else:
             ManagedProcess.spawn(["dvc"] + cmd, wdir=self.wdir, name=name)
 
+        for _ in range(5):
+            time.sleep(1)
+            if node_name in self.active_worker():
```

Comment:
The newly created node can only be detected some time later.

Reply:
I don't think we need this check at all here. We already expect that in some cases the new worker may not start at all (if the node name is already in use it should exit immediately).